### PR TITLE
refactor universal size-amp picking

### PR DESCRIPTION
**Pull request by ajkr**
There were three loops over `sorted_runs` where only one is needed. We just need to iterate backwards over the vector instead of forwards since we're looking for a span of sorted runs including the final element.

Also I am experimenting with some changes to size-amp compaction that depend on this change.

Test Plan: unit tests

### CODE REVIEW

File: db/compaction_picker_universal.cc [hunk 1]
```diff
@@ -614,66 +614,46 @@ Compaction* UniversalCompactionPicker::PickCompactionToReduceSortedRuns(
...
       break;
     }
-    char file_num_buf[kFormatFileNumberBufSize];
-    sr->Dump(file_num_buf, sizeof(file_num_buf), true);
-    ROCKS_LOG_BUFFER(log_buffer, "[%s] Universal: skipping %s[%d] compacted %s",
-                     cf_name.c_str(), file_num_buf, loop,
-                     " cannot be a candidate to reduce size amp.\n");
-    sr = nullptr;
+    candidate_size += sr->compensated_file_size;
+    estimated_total_size += sr->size;
```
Comment (siying) [inline db/compaction_picker_universal.cc]: Did I miss something? `estimated_total_size` now only ends at index `sorted_runs.size() - 2`, and before the change, it include the last file?

Comment (ajkr) [inline db/compaction_picker_universal.cc]: Oh, that's a really good point, let me fix it.

Comment (ajkr) [inline db/compaction_picker_universal.cc]: I verified the value of `estimated_total_size` is the same before and after this change for a unit test that does a lot of size-amp compactions:
```
TEST_TMPDIR=/dev/shm ./db_universal_compaction_test --gtest_filter=UniversalCompactionNumLevels/DBTestUniversalCompaction.UniversalCompactionSingleSortedRun/0
```

**Comment by gfosco** [comment]
Closing this via automation due to lack of activity.  If you'd like this to be re-opened, please just comment here and we'll open it back up.

**Pull request closed without merging**